wifiphisher.py

```diff
@@ -881,32 +884,127 @@ def get_hostapd():
     # Get hostapd if needed
     get_hostapd()
 
+    # get all the local templates
+    local_templates = phishingpage.get_local_templates()
+
+    # loop through the local templates
+    for name in local_templates:
+
+        # add the template to the database
+        phishingpage.add_template(name)
+
     # get template_database
     template_database = phishingpage.get_template_database()
 
-    # check to see if the template is local
-    if not template_database[TEMPLATE_NAME] is None:
+    # get all the templtes names for display
+    template_names = list(template_database.keys())
+
+    # variable used to indicate correct template selection
+    template_selected = False
+
+    # loop until all operations for template selection is done
+    while True:
+
+        # check if the template argument is set and is correct
+        if args.template and args.template in template_database:
+
+            # set the template name
+            template_name = args.template
+
+            # set the template selected to avoid interactive selection
+            template_selected = True
```

Comment:
Before implementing the template argument, I set it up so if you chose an online template and then press no to download it would come back to the start. However after some thought I think it would be best to raise an exception if the argument is wrong instead of allowing it to continue.